// mongodb-cypress/dist/src/functions.js
"use strict";
/// <reference types="cypress" />
/// <reference path="../types/index.d.ts" />
Object.defineProperty(exports, "__esModule", { value: true });
exports.functions = void 0;
var functions;
(function (functions) {
    functions[functions["mongoFindOne"] = 0] = "mongoFindOne";
    functions[functions["mongoFindMany"] = 1] = "mongoFindMany";
    functions[functions["mongoInsertOne"] = 2] = "mongoInsertOne";
    functions[functions["mongoInsertMany"] = 3] = "mongoInsertMany";
    functions[functions["mongoDeleteMany"] = 4] = "mongoDeleteMany";
    functions[functions["mongoDeleteOne"] = 5] = "mongoDeleteOne";
    functions[functions["mongoUpdateMany"] = 6] = "mongoUpdateMany";
    functions[functions["mongoUpdateOne"] = 7] = "mongoUpdateOne";
})(functions || (exports.functions = functions = {}));
module.exports = function () {
    Cypress.Commands.add("mongoFindMany", function (collection, filter, options) {
        return cy.task("mongoConnection", {
            fun: functions.mongoFindMany,
            collection: collection,
            findParameters: { filter: filter, options: options },
        });
    });
    Cypress.Commands.add("mongoFindOne", function (collection, filter, options) {
        return cy.task("mongoConnection", {
            fun: functions.mongoFindOne,
            collection: collection,
            findParameters: { filter: filter, options: options },
        });
    });
    Cypress.Commands.add("mongoInsertOne", function (collection, item, options) {
        return cy.task("mongoConnection", {
            fun: functions.mongoInsertOne,
            collection: collection,
            insertParameters: { item: item, options: options },
        });
    });
    Cypress.Commands.add("mongoInsertMany", function (collection, item, options) {
        return cy.task("mongoConnection", {
            fun: functions.mongoInsertMany,
            collection: collection,
            insertParameters: { item: item, options: options },
        });
    });
    Cypress.Commands.add("mongoDeleteMany", function (collection, filter) {
        return cy.task("mongoConnection", {
            fun: functions.mongoDeleteMany,
            collection: collection,
            deleteParameters: { filter: filter },
        });
    });
    Cypress.Commands.add("mongoDeleteOne", function (collection, filter) {
        return cy.task("mongoConnection", {
            fun: functions.mongoDeleteOne,
            collection: collection,
            deleteParameters: { filter: filter },
        });
    });
    Cypress.Commands.add("mongoUpdateMany", function (collection, filter, update) {
        return cy.task("mongoConnection", {
            fun: functions.mongoUpdateMany,
            collection: collection,
            updateParameters: { filter: filter, update: update },
        });
    });
    Cypress.Commands.add("mongoUpdateOne", function (collection, filter, update) {
        return cy.task("mongoConnection", {
            fun: functions.mongoUpdateOne,
            collection: collection,
            updateParameters: { filter: filter, update: update },
        });
    });
};
